plugins/outputs/opentelemetry/opentelemetry.go

```diff
@@ -0,0 +1,182 @@
+package opentelemetry
+
+import (
+	"context"
+	"time"
+
+	"github.com/influxdata/influxdb-observability/common"
+	"github.com/influxdata/influxdb-observability/influx2otel"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"go.opentelemetry.io/collector/model/otlpgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+type OpenTelemetry struct {
+	ServiceAddress string `toml:"service_address"`
+
+	tls.ClientConfig
+	Timeout     config.Duration   `toml:"timeout"`
+	Compression string            `toml:"compression"`
+	Headers     map[string]string `toml:"headers"`
```

Comment:
I'm not seeing where this is currently used, these headers should be setting grpc metadata in the client connections. See https://github.com/influxdata/telegraf/pull/9109/files#diff-2702903f71c7896e7e1d358e50075b5dc48150603438ee5eaea24948b8230d18R111